packages/api/cms-api/src/content-generation/content-generation.config.ts

```diff
@@ -0,0 +1,7 @@
+export type OpenAiContentGenerationConfig<T> = {
+    [K in keyof T]: {
```

Comment:
What's your intention behind this?

Reply:
I want to automatically create the type of the config by extracting the function names of the `ContentGenerationServiceInterface` and each having `deploymentId: string; apiKey: string; apiUrl: string;` as properties. In this example, it should look like this: 
```
{
    generateAltText: { deploymentId: string; apiKey: string; apiUrl: string };
    generateImageTitle: { deploymentId: string; apiKey: string; apiUrl: string };
}
```

Reply:
But won't the provided values always be the same?